chromadb/ingest/__init__.py

```diff
@@ -37,14 +71,42 @@ def reset(self) -> None:
         pass
 
 
+ConsumerCallbackFn = Callable[
+    [Sequence[Union[EmbeddingRecord, EmbeddingDeleteRecord]]], None
+]
+
+
+class RejectedEmbeddingException(Exception):
```

Comment:
I believe using exceptions to handle rejected embeddings is the right direction, since that allows us to "unwind" any failures including rolling back transactions when executed in a synchronous context.

However, before we merge this, I want to make sure this works at a conceptual level for the distributed system version. Working on an ADR now and will include a link to that.